src/main/java/br/com/fiap/tech_challenge/adapters/driven/infra/repository/ProductPersistenceImpl.java

```diff
@@ -24,10 +30,25 @@ public Product create(Product product) {
         return productSaved.toProduct();
     }
 
+    public Page<Product> findByCategory(CategoryProductEnum category, Pageable pageable) {
+        return repository.findByCategoryAndStatusNot(category, INACTIVE, pageable).map(ProductEntity::toProduct);
+    }
+
     @Override
+    public Optional<Product> findById(UUID id) {
+        return repository.findByIdAndStatusNot(id, INACTIVE).map(ProductEntity::toProduct);
```

Comment:
Mano, tem um jeito melhor de lidar com essa parada de GET quando estamos trabalhando com deleção lógica, da uma olhada nesse video falando sobre a anotação @Where utilizada na classe entidade.
https://youtu.be/-EYXYci9kVI?si=RJBYSvRLMdswM6vK&t=850

Reply:
Fechou!! Obrigado mano!